multy_test/test_golos_transaction.cpp

```diff
@@ -0,0 +1,265 @@
+ * Licensed under Multy.io license.
+ *
+ * See LICENSE for details
+ */
+
+#include "multy_core/src/golos/golos_account.h"
+#include "multy_core/binary_data.h"
+
+#include "multy_core/account.h"
+#include "multy_core/big_int.h"
+#include "multy_core/properties.h"
+#include "multy_core/transaction.h"
+
+#include "multy_core/src/api/big_int_impl.h"
+#include "multy_core/src/api/account_impl.h"
+#include "multy_core/src/api/key_impl.h"
+#include "multy_core/src/u_ptr.h"
+#include "multy_core/src/utility.h"
+
+#include "multy_test/utility.h"
+#include "multy_test/value_printers.h"
+
+#include "multy_core/golos.h"
+
+#include "gtest/gtest.h"
+
+namespace
+{
+using namespace multy_core::internal;
+using namespace test_utility;
+} // namespace
+
+GTEST_TEST(GolosTransactionTest, SmokeTest_public_api)
+{
+    AccountPtr account;
+
+    HANDLE_ERROR(make_account(
+                     BLOCKCHAIN_GOLOS,
+                     "5JpDgood17pE47zB6pDJixg9Sw47QiHcQ9qCc3MeKYoYzRiMcnF",
+                     reset_sp(account)));
+
+    TransactionPtr transaction;
+    HANDLE_ERROR(make_transaction(account.get(), reset_sp(transaction)));
+    ASSERT_NE(nullptr, transaction);
+
+    {
+        Properties* properties = nullptr;
+        const BigInt ref_block_num(14918824);
+        BinaryDataPtr ref_block_prefix;
+        const BigInt expiration(1521729720); // 2018-03-22T14:42:00
+
+        HANDLE_ERROR(make_binary_data_from_hex("0x00e3a4a84407f2df4953c35614248b433e6db43e", reset_sp(ref_block_prefix)));
+
+        HANDLE_ERROR(transaction_get_properties(transaction.get(), &properties));
+        HANDLE_ERROR(properties_set_big_int_value(properties, "ref_block_num", &ref_block_num));
+        HANDLE_ERROR(properties_set_binary_data_value(properties, "ref_block_prefix", ref_block_prefix.get()));
+        HANDLE_ERROR(properties_set_big_int_value(properties, "expiration", &expiration));
+    }
+
+    {
+        Properties* source = nullptr;
+        HANDLE_ERROR(transaction_add_source(transaction.get(), &source));
+
+        // Address balance
+        BigIntPtr balance;
+        HANDLE_ERROR(make_big_int("2000", reset_sp(balance))); // balance = 2.000 GOLOS
+
+        HANDLE_ERROR(properties_set_big_int_value(source, "amount", balance.get()));
+        HANDLE_ERROR(properties_set_string_value(source, "from", "multytest"));
+    }
+
+    {
+        Properties* destination = nullptr;
+        HANDLE_ERROR(transaction_add_destination(transaction.get(), &destination));
+
+        BigIntPtr amount;
+        HANDLE_ERROR(make_big_int("5", reset_sp(amount))); // amount = 0.005 GOLOS
+        HANDLE_ERROR(properties_set_big_int_value(destination, "amount", amount.get()));
+        HANDLE_ERROR(properties_set_string_value(destination, "to", "multy"));
+    }
+
+    // transaction_id = "eb096e9f1e4a34c0d9b72c3193b8a77bc3480a01"
+    BinaryDataPtr signatures;
+    HANDLE_ERROR(transaction_serialize(transaction.get(), reset_sp(signatures)));
+    ASSERT_EQ(as_binary_data(R"({
```

Comment:
Please add a function that strips away all whitespace in given JSON string (except for quoted whitespace). Use that  for both JSON literal and serialized TX.